reflex/utils/exec.py

```diff
@@ -248,8 +248,17 @@ def get_reload_dirs() -> list[str]:
     """
     config = get_config()
     reload_dirs = [config.app_name]
-    if app_module_path := config.app_module_path:
-        reload_dirs.append(str(Path(app_module_path).resolve().parent.parent))
+    if config.app_module is not None and config.app_module.__file__:
```

Comment:
I initially based my assumption on the current behavior of how we expect reflex apps to be structured based on the  `Config.module` property(`return ".".join([self.app_name, self.app_name])`). This actually assumes that the entrypoint (or where the App instance lives) will only be one level deep, So if I modify my file structure from `foo/foo/foo.py` to `foo/foo/bar/foo.py` this will currently fail. With this new approach, I'm wondering if we need to have a similar behavior where we provide users with a knob to change the default behavior, but I'm also wondering if adding another knob will make it confusing for users since we have the `app_module_import` knob now

Reply:
i don't think we necessarily need to do anything special here. we just want to watch the top of the package hierarchy that the `app_import_module` belongs to. In fact the same logic would work just fine with the "default" structure as well.